app/controllers/users_controller.rb

```diff
@@ -1,3 +1,33 @@
 class UsersController < ApplicationController
-  def new; end
+  def show
+    @user = User.find_by id: params[:id]
+    return if @user
+
+    flash[:warning] = t ".user_not_found"
+    redirect_to root_path
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new user_params
+    if @user.save
+      reset_session
```

Comment:
reset_session hàm này làm gì e

Reply:
`reset_session` là làm mới lại phiên, dùng để xoá toàn bộ dữ liệu cũ, làm mới dữ liệu hiện tại của người dùng trong phiên làm việc đó để tránh các cuộc tấn công khi hacker biết trước id của phiên làm việc cũ.